fetcher/handlers/l0_fetcher.py

```diff
@@ -47,21 +63,58 @@ def format_command(
     return cmd
 
 
+def parse_log(log: str) -> List[str]:
+    def keep(row: Dict[str, str]) -> bool:
+        return row["msg"] != "Deleted"
+
+    return [
+        parsed_row["object"]
+        for parsed_row in [json.loads(row) for row in log.splitlines()]
+        if "object" in parsed_row and keep(parsed_row)
+    ]
+
+
+def notify_queue(
+    sqs_client: BotoClient,
+    notification_queue: str,
+    modified_files: List[str],
+    bucket: str,
+) -> None:
+    response = sqs_client.send_message(
+        QueueUrl=notification_queue,
+        MessageBody=json.dumps({
+            "objects": modified_files,
+            "bucket": bucket,
+        })
+    )
+    if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        msg = f"Failed to notify queue: {response}"
+        logging.error(msg)
+        raise SyncError(msg)
+
+
 def lambda_handler(event: Event, context: Context):
     ssm_client: BotoClient = boto3.client("ssm")
+    sqs_client: BotoClient = boto3.client("sqs")
 
     config_path = get_rclone_config_path(
         ssm_client,
         os.environ.get("RCLONE_CONFIG_SSM_NAME", "config")
     )
-    bucket = os.environ.get("OUTPUT_BUCKET", "bucket")
+    bucket = get_or_raise("OUTPUT_BUCKET")
+    source_path = get_or_raise("SOURCE_PATH")
+    notification_queue = get_or_raise("NOTIFICATION_QUEUE")
     full_sync = os.environ.get("FULL_SYNC", "FALSE").upper() != "FALSE"
-    source_path = os.environ.get("SOURCE_PATH", "/")
 
     cmd = format_command(config_path, source_path, bucket, full_sync)
 
     out = subprocess.run(cmd, capture_output=True)
-    if out.returncode == 0:
-        logging.info("Files successfully synced")
-    else:
+    if out.returncode != 0:
         logging.error(out.stderr)
+        raise SyncError(out.stderr.decode())
+
+    logging.info("Files successfully synced")
```

Comment:
I am pretty sure they are available in Cloudwatch "forever".

Hmm... I think I'm leaning more towards removing all this logging rather than fleshing it out. And then if we realize it would be useful to know how many files are synced or whatnot we add it later. What do you think?

Reply:
Good idea. Then we can make an active decision if we want to log with metrics in mind.